libraries/AP_HAL_ChibiOS/hwdef/JFB100/hwdef.dat

```diff
@@ -0,0 +1,354 @@
+SPIDEV scha63t_g      SPI1 DEVID1  SCHA63T_G_CS MODE0 10*MHZ 10*MHZ
+SPIDEV scha63t_a      SPI1 DEVID2  SCHA63T_A_CS MODE0 10*MHZ 10*MHZ
+SPIDEV icm20602       SPI1 DEVID3  ICM20602_CS  MODE3  2*MHZ  8*MHZ
+SPIDEV ramtron        SPI2 DEVID1  FRAM_CS      MODE3  8*MHZ  8*MHZ 
+
+# up to 3 IMUs
+IMU SCHA63T SPI:scha63t_a SPI:scha63t_g ROTATION_NONE
+IMU Invensense SPI:icm20602 ROTATION_PITCH_180_YAW_270
+
+# 3rd could be BMMI055 or BMI088
+# IMU BMI055 SPI:bmi055_a SPI:bmi055_g ROTATION_ROLL_180_YAW_90
+# IMU BMI088 SPI:bmi055_a SPI:bmi055_g ROTATION_ROLL_180_YAW_90
+
+define HAL_DEFAULT_INS_FAST_SAMPLE 1
+
+# probe external I2C compasses plus some internal IST8310
+# we also probe some external IST8310 with a non-standard orientation
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+COMPASS IST8310 I2C:ALL_EXTERNAL:0x0E true  ROTATION_ROLL_180_YAW_90
+COMPASS IST8310 I2C:ALL_INTERNAL:0x0E false ROTATION_YAW_270
+
+# one baro
+BARO MS56XX SPI:ms5611
+
+# microSD support
+PC8 SDMMC_D0 SDMMC1
+PC9 SDMMC_D1 SDMMC1
+PC10 SDMMC_D2 SDMMC1
+PC11 SDMMC_D3 SDMMC1
+PC12 SDMMC_CK SDMMC1
+PD2 SDMMC_CMD SDMMC1
+
+# red LED marked as B/E
+# PB1 LED_RED OUTPUT OPENDRAIN GPIO(90) (swap)
+
+# green LED marked as PWR. We leave this solid on, but allow
+# for it to be controlled as a relay if needed
+PC6 LED_GREEN OUTPUT GPIO(91) LOW
+
+# blue LED marked as ACT
+# PC7 LED_BLUE OUTPUT GPIO(92) HIGH (swap)
+
+# setup for BoardLED2
+define HAL_GPIO_A_LED_PIN 90
+define HAL_GPIO_B_LED_PIN 92
+define HAL_GPIO_LED_ON 0
+
+
+# enable RAMTROM parameter storage
+define HAL_STORAGE_SIZE 32768
+define HAL_WITH_RAMTRON 1
+
+# allow to have have a dedicated safety switch pin
+define HAL_HAVE_SAFETY_SWITCH 1
+
+define HAL_COMPASS_DEFAULT HAL_COMPASS_NONE
+define HAL_COMPASS_AUTO_ROT_DEFAULT 2
+
+DMA_PRIORITY SDMMC* UART8* ADC* SPI* TIM*
+
+# define HAL_SPI_CHECK_CLOCK_FREQ 1
+
+# enable FAT filesystem support (needs a microSD defined via SDMMC)
+define HAL_OS_FATFS_IO 1
+
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+
+# uncomment block below to enable a 2nd MS5611 baro on SPI5
+#PF7 SPI5_SCK SPI5
+#PF8 SPI5_MISO SPI5
+#PF9 SPI5_MOSI SPI5
+#SPIDEV ms5611_spi5    SPI5 DEVID1  EXTERNAL1_CS1 MODE3 20*MHZ 20*MHZ
+#undef PI10
+#PI10 EXTERNAL1_CS2 OUTPUT LOW
+#BARO MS56XX SPI:ms5611_spi5
+
+# imu sensor SCHA63T reset
+PH12 SCHA63T_RESET OUTPUT LOW
+
+# PWM wraparound 1-8
+PA8 WA1 INPUT PULLUP
+PA15 WA2 INPUT PULLUP
+PB2 WA3 INPUT PULLUP
+PB10 WA4 INPUT PULLUP
+PE7 WA5 INPUT PULLUP
+PF11 WA6 INPUT PULLUP
+PG8 WA7 INPUT PULLUP
+PG15 WA8 INPUT PULLUP
+
+# now we define the pins that USB is connected on
+PA9 VBAS INPUT PULLDOWN
+
+# External watchdog timer clear
+PH11 WDI_OUT OUTPUT SPEED_VERYLOW
+define WDI_OUT_INTERVAL_TIME_MS 100
+
+# safety switch
+PE10 SAFETY_IN INPUT PULLUP
+
+# red and blue LEDs swapped vs fmuv5
+# red LED marked as B/E
+PB1 LED_RED OUTPUT OPENDRAIN GPIO(92)
```

Comment:
Conflict for ping 90 and 92?

Reply:
the other definitions that use GPIO(90) and GPIO(92) are actually commented out so no actual conflict.

Reply:
I am sorry for confusing you.
JFB100 board is used GPIO(90) set to LED_BLUE and GPIO(92) set to LED_RED.
I will corrected hwdef to make it easier to understand.

Reply:
I cannot modify the source code in this PR.
I put the modified hwdef in the following, so please use it.
https://github.com/jfbblue0922/JFB_ardupilot/blob/JFB100-new/libraries/AP_HAL_ChibiOS/hwdef/JFB100/hwdef.dat

Reply:
@jfbblue0922,

Thanks very much for that.  I've added a new commit with the LED fix.

By the way, I see your hwdef file has a custom firmware string.  This is not really a great idea because it hides the ArduPilot version so I've removed it.